tests/output.rs

```diff
@@ -550,3 +572,75 @@ fn test_filter_plugin_unparseable_argument() {
         "#,
     );
 }
+
+#[test]
+fn test_filter_plugin_no_query() {
+    build_example("filter_function_items_before_line_10");
+
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --language rust --filter ../../../target/debug/examples/libfilter_function_items_before_line_10.so
+            src/helpers.rs:1:pub fn helper() {}
+            src/lib.rs:3:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:4:    left + right
+            src/lib.rs:5:}
+            src/stop.rs:1:fn stop_it() {}
+        "#,
+    );
+}
+
+#[test]
+fn test_query_inline_and_query_file_path() {
+    assert_failure_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-source '(function_item) @function_item' --query-file ./function-item.scm --language rust
+            error: the argument '--query-source <QUERY_SOURCE>' cannot be used with '--query-file <PATH_TO_QUERY_FILE>'
+
+            Usage: tree-sitter-grep --language <LANGUAGE> <--query-file <PATH_TO_QUERY_FILE>|--query-source <QUERY_SOURCE>|--filter <FILTER>> [PATHS]...
+
+            For more information, try '--help'.
+        "#,
+    );
+}
+
+#[test]
+fn test_help_option() {
+    assert_non_match_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --help
+            Usage: tree-sitter-grep [OPTIONS] <--query-file <PATH_TO_QUERY_FILE>|--query-source <QUERY_SOURCE>|--filter <FILTER>> [PATHS]...
```

Comment:
While this could be considered technically slightly misleading (it makes it look like `--filter` is mutually exclusive with the query arguments) it's pretty close and I'm not really sure how you'd even write the actual logic out in a format like this?